packages/grafana-e2e-selectors/src/selectors/index.ts

```diff
@@ -1,21 +1,24 @@
+import { resolveSelectors } from '../resolver';
 import { E2ESelectors } from '../types';
 
-import { Components } from './components';
-import { Pages } from './pages';
+import { versionedComponents, VersionedComponents } from './components';
+import { versionedPages, VersionedPages } from './pages';
 
-/**
- * Exposes selectors in package for easy use in e2e tests and in production code
- *
- * @alpha
- */
-export const selectors: { pages: E2ESelectors<typeof Pages>; components: E2ESelectors<typeof Components> } = {
-  pages: Pages,
-  components: Components,
-};
+const Pages = resolveSelectors<VersionedPages>(versionedPages);
+const Components = resolveSelectors<VersionedComponents>(versionedComponents);
```

Comment:
given that `VersionedPages`/`VersionedComponents` are just `typeof versionedPages`/`typeof versionedComponents` anyway, couldn't we just infer these?

e.g. `resolveSelectors` would become:
```tsx
export function resolveSelectors<T extends VersionedSelectorGroup>(
  versionedSelectors: T,
  grafanaVersion = 'latest'
): SelectorsOf<T> {
```

and this would become
```tsx
const Pages = resolveSelectors(versionedPages);
const Components = resolveSelectors(versionedComponents);
```

Reply:
Good shout, thanks!